aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -3427,8 +3429,33 @@ static void linalg_lstsq_out_info(
   auto input_working_copy = copyBatchedColumnMajor(input);
 
   // now the actual call that computes the result in-place (apply_lstsq)
-  lstsq_stub(input.device().type(), input_working_copy, solution, rank, singular_values, infos, rcond, driver);
-
+  if(driver == "gelsd" || driver == "gelss") {
+    auto k = std::min(m, n);
+    auto U = at::zeros({input.size(0), m, k}, input.options());
+    auto Vh = at::zeros({input.size(0), k, n}, input.options());
```

Comment:
zeros actually zeros out `U` and `Vh`. We don't need that. Rather than going so low level, you can call the higher level `at::_linalg_svd`.